inspire_disambiguation/api.py

```diff
@@ -21,127 +21,162 @@
+    curated_signatures = get_signatures(only_curated=True)
+    input_clusters = get_input_clusters(curated_signatures)
+    LOGGER.info(
+        "Preparing %s pairs from sampled data for training.", sampled_pairs_size
     )
+    pairs = [
+        pair
+        for pair in sample_signature_pairs(
+            curated_signatures, input_clusters, sampled_pairs_size
+        )
+    ]
+
+    ethnicity_estimator = EthnicityEstimator(ethnicity_model_path)
+    distance_estimator = DistanceEstimator(ethnicity_estimator)
+    distance_estimator.load_data(curated_signatures, pairs, sampled_pairs_size)
+    LOGGER.info("Training DistanceEstimator...")
     distance_estimator.fit()
-    distance_estimator.save_model(current_app.config['DISAMBIGUATION_DISTANCE_MODEL_PATH'])
+    distance_estimator.save_model(save_distance_model_path)
+
 
+def cluster(ethnicity_model_path, distance_model_path, n_jobs, signature_block=None):
+    """Train the clustering model and process the output.
 
-def train_and_save_clustering_model():
-    """Train the clustering model and save it to disk."""
-    ethnicity_estimator = EthnicityEstimator()
-    ethnicity_estimator.load_model(current_app.config['DISAMBIGUATION_ETHNICITY_MODEL_PATH'])
+    Args:
+        ethnicity_model_path (str): Full path where ethnicity model is saved.
+        distance_model_path (str): Full path where distance model is saved.
+        n_jobs (int): Number of processes to use.
+        signature_block (str): Signature block indicating which block should be
+            clustered. If set to None, clustering will run on all blocks.
+    """
+    LOGGER.info(
+        "Pulling requested signatures and input_clusters ('%s') form ES",
+        signature_block,
+    )
 
+    signatures = get_signatures(signature_block=signature_block)
+    input_clusters = get_input_clusters(signatures)
+    LOGGER.debug(
+        "Got %s signature_blocks and %s input_clusters",
+        len(signatures),
+        len(input_clusters),
+    )
+    ethnicity_estimator = EthnicityEstimator(ethnicity_model_path)
     distance_estimator = DistanceEstimator(ethnicity_estimator)
-    distance_estimator.load_model(current_app.config['DISAMBIGUATION_DISTANCE_MODEL_PATH'])
+    distance_estimator.load_model(distance_model_path)
 
     clusterer = Clusterer(distance_estimator)
-    clusterer.load_data(
-        current_app.config['DISAMBIGUATION_CURATED_SIGNATURES_PATH'],
-        current_app.config['DISAMBIGUATION_PUBLICATIONS_PATH'],
-        current_app.config['DISAMBIGUATION_INPUT_CLUSTERS_PATH'],
-    )
-    clusterer.fit(n_jobs=current_app.config['DISAMBIGUATION_CLUSTERING_N_JOBS'])
-    clusterer.save_model(current_app.config['DISAMBIGUATION_CLUSTERING_MODEL_PATH'])
+    clusterer.load_data(signatures, input_clusters)
+    LOGGER.info("Starting clustering")
+    clusterer.fit(n_jobs=n_jobs)
+
+    return process_clustering_output(clusterer)
+
 
+def process_clustering_output(clusterer):
```

Comment:
maybe move this somewhere else? it's not part of the API.

Reply:
I believe we can do this when output will be more stable.